In [14]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import scipy.stats as st
import random
import seaborn as sns
import math
import statistics

Задание 1:

In [ ]:
task_1 = pd.read_csv('Task_1.csv')
print(len(task_1))

task_1.head(3)

In [127]:
m = np.round(np.mean(task_1['Values']), 2)
s = np.round(np.std(task_1['Values']), 2)
var = np.round(statistics.variance((task_1['Values'])), 2)
mode = statistics.mode(task_1['Values'])
print('Среднее равно: ', m)
print('Стандартное отклонение равно:', s)
print('Дисперсия равна: ', var)
print('Мода равна: ', round(mode, 2))

Среднее равно:  177.34
Стандартное отклонение равно: 115.23
Дисперсия равна:  13285.81
Мода равна:  81.38


In [126]:
perc_2 = np.percentile(task_1['Values'], 2)
print("2 процентиль равен:", round(perc_2, 2))

2 процентиль равен: 38.94


In [125]:
quant_1 = np.quantile(task_1['Values'], 0.25)
quant_2 = np.quantile(task_1['Values'], 0.5)
quant_3 = np.quantile(task_1['Values'], 0.75)
quant_4 = np.quantile(task_1['Values'], 1)
print("1-й квартиль равен:", round(quant_1, 2))
print("2-й квартиль равен:", round(quant_2, 2))
print("3-й квартиль равен:", round(quant_3, 2))
print("4-й квартиль равен:", round(quant_4, 2))

1-й квартиль равен: 99.23
2-й квартиль равен: 151.07
3-й квартиль равен: 220.37
4-й квартиль равен: 928.66


In [124]:
iqr = quant_3 - quant_1
print("Интерквартильный размах равен:", round(iqr, 2))

Интерквартильный размах равен: 121.14


Задание 2.1:

In [ ]:
task_2_1 = pd.read_csv('Task_2_1.csv')
print(len(task_2_1))

task_2_1.head(3)

In [122]:
m2 = np.round(np.mean(task_2_1['Values']), 2)
s2 = np.round(np.std(task_2_1['Values']), 2)
var2 = np.round(statistics.variance((task_2_1['Values'])), 2)

print('Среднее равно: ', m2)
print('Стандартное отклонение равно:', s2)
print('Дисперсия равна: ', var2)

Среднее равно:  877.68
Стандартное отклонение равно: 656.45
Дисперсия равна:  436753.42


In [121]:
quant_1_task2 = np.quantile(task_2_1['Values'], 0.25)
quant_3_task2 = np.quantile(task_2_1['Values'], 0.75)
print("1-й квартиль равен:", round(quant_1_task2, 2))
print("3-й квартиль равен:", round(quant_3_task2, 2))


1-й квартиль равен: 296.66
3-й квартиль равен: 1244.9


Задание 2.2: 

In [ ]:
task_2_2 = pd.read_csv('Task_2_2.csv')
print(len(task_2_2))

task_2_2.head(3)

In [120]:
# абсолютные частоты
task_2_2['Value'].value_counts()

Purple Audi      30
Red Mercedes     27
Blue BMW         23
Jellow Toyota    22
Name: Value, dtype: int64

In [119]:
# относительные частоты
task_2_2['Value'].value_counts(normalize=True).round(2)

Purple Audi      0.29
Red Mercedes     0.26
Blue BMW         0.23
Jellow Toyota    0.22
Name: Value, dtype: float64

In [65]:
task_2_2['Value'].value_counts().index[-1]

'Jellow Toyota'

Задание 3:

In [ ]:
task_2_1 = pd.read_csv('Task_2_1.csv')
print(len(task_2_1))

task_2_1.head(3)

In [103]:
results = []
for i in range(10_000):
    sample1 = np.random.choice(task_2_1['Values'], 75, replace=True)
    results.append(sample1.mean())

In [ ]:
sns.distplot(results)

In [108]:
#t-распределение
CI_t = st.t.interval(alpha=0.9, df=len(results)-1, loc=np.mean(results), scale=np.std(results))
print("90% доверительный интервал среднего (t-распределение):", CI_t)

90% доверительный интервал среднего (t-распределение): (753.9047285804638, 1001.346585606203)


C:\Users\Марина\AppData\Local\Temp\ipykernel_15480\2889623002.py:2: DeprecationWarning: Use of keyword argument `alpha` for method `interval` is deprecated. Use first positional argument or keyword argument `confidence` instead.
  CI_t = st.t.interval(alpha=0.9, df=len(results)-1, loc=np.mean(results), scale=np.std(results))


Задание 4:

Истинное среднее нашей совокупности (783.21) попадает в доверительный интервал выборки (753.9 - 1001.3)
Сгенерировать выборку для проведения доп. тестов не можем, т.к неизвестны  распределения, средние, дисперсии нашей совокупности и т.д.

Задание 5:

In [112]:
from scipy import integrate

In [113]:
def func(x):   
    return (
        1/(
            7.5*np.sqrt(2*np.pi)
            ))*np.exp(-(x-32)**2/(2*7.5**2)
                      )*(x-32)**2/(2*7.5**2)

In [115]:
integrate.quad(func,-40,35)[0]

0.25405684274449747